src/CMakeLists.txt

```diff
@@ -0,0 +1,659 @@
+
+if(USE_AOT_EXECUTOR)
+  message(STATUS "Build with AOT Executor support...")
+  file(GLOB RUNTIME_AOT_EXECUTOR_SRCS runtime/aot_executor/*.cc)
+  list(APPEND RUNTIME_SRCS ${RUNTIME_AOT_EXECUTOR_SRCS})
+
+endif(USE_AOT_EXECUTOR)
+
+if(USE_PIPELINE_EXECUTOR)
+  message(STATUS "Build with Pipeline Executor support...")
+  tvm_file_glob(GLOB RUNTIME_PIPELINE_SRCS runtime/pipeline/*.cc)
+  list(APPEND RUNTIME_SRCS ${RUNTIME_PIPELINE_SRCS})
+endif(USE_PIPELINE_EXECUTOR)
+
+# Module rules
+include(../cmake/modules/VTA.cmake)
+include(../cmake/modules/StandaloneCrt.cmake)
+include(../cmake/modules/Zephyr.cmake)
+include(../cmake/modules/Arduino.cmake)
+include(../cmake/modules/CUDA.cmake)
+include(../cmake/modules/Hexagon.cmake)
+include(../cmake/modules/OpenCL.cmake)
+include(../cmake/modules/OpenMP.cmake)
+include(../cmake/modules/Vulkan.cmake)
+include(../cmake/modules/Metal.cmake)
+include(../cmake/modules/ROCM.cmake)
+include(../cmake/modules/LLVM.cmake)
+include(../cmake/modules/Micro.cmake)
+include(../cmake/modules/contrib/EthosN.cmake)
+include(../cmake/modules/contrib/CMSISNN.cmake)
+include(../cmake/modules/contrib/EthosU.cmake)
+include(../cmake/modules/contrib/BLAS.cmake)
+include(../cmake/modules/contrib/CODEGENC.cmake)
+include(../cmake/modules/contrib/DNNL.cmake)
+include(../cmake/modules/contrib/CUTLASS.cmake)
+include(../cmake/modules/contrib/ExampleTargetHooks.cmake)
+include(../cmake/modules/contrib/Random.cmake)
+include(../cmake/modules/contrib/Posit.cmake)
+include(../cmake/modules/contrib/MicroStandaloneRuntime.cmake)
+include(../cmake/modules/contrib/Sort.cmake)
+include(../cmake/modules/contrib/NNPack.cmake)
+include(../cmake/modules/contrib/HybridDump.cmake)
+include(../cmake/modules/contrib/TFLite.cmake)
+include(../cmake/modules/contrib/TF_TVMDSOOP.cmake)
+include(../cmake/modules/contrib/PT_TVMDSOOP.cmake)
+include(../cmake/modules/contrib/CoreML.cmake)
+include(../cmake/modules/contrib/BNNS.cmake)
+include(../cmake/modules/contrib/ONNX.cmake)
+include(../cmake/modules/contrib/ArmComputeLib.cmake)
+include(../cmake/modules/contrib/TensorRT.cmake)
+include(../cmake/modules/contrib/VitisAI.cmake)
+include(../cmake/modules/contrib/Verilator.cmake)
+include(../cmake/modules/Git.cmake)
+include(../cmake/modules/LibInfo.cmake)
+include(../cmake/modules/RustExt.cmake)
+
+if (PROJECT_IS_TOP_LEVEL)
+    include(CheckCXXCompilerFlag)
+    if(NOT MSVC)
+        check_cxx_compiler_flag("-std=c++14" SUPPORT_CXX14)
+        set(CMAKE_CXX_FLAGS "-std=c++14 ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CUDA_STANDARD 14)
+    else()
+        check_cxx_compiler_flag("/std:c++14" SUPPORT_CXX14)
+        set(CMAKE_CXX_FLAGS "/std:c++14 ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CUDA_STANDARD 14)
+    endif()
+endif ()
+
+set(LIBINFO_FILE ${CMAKE_CURRENT_LIST_DIR}/support/libinfo.cc)
+add_lib_info(${LIBINFO_FILE})
+list(REMOVE_ITEM COMPILER_SRCS ${LIBINFO_FILE})
+
+add_library(${PRJ_NAME} SHARED ${COMPILER_SRCS} ${RUNTIME_SRCS} ${LIBINFO_FILE})
```

Comment:
The reason why `tvm_runtime_objs` and `tvm_objs` exist is so that we do not compile files in the runtime twice. You will need to add them back in.